### PR TITLE
Replace Closure and Flambda 1 CI jobs by a Flambda 2 + debug runtime5 CI job

### PR DESCRIPTION
**Pull request by mshinwell**
As subject.

### CODE REVIEW

**Comment by riaqn** [comment]
Now that #2118 is merged, this PR should pass CI after rebase.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -13,27 +13,16 @@ jobs:
...
+          - name: flambda2_runtime5
+            config: --enable-middle-end=flambda2 --enable-runtime5
             os: ubuntu-latest
-            build_ocamlparam: ''
 
-          - name: flambda2_runtime5
+          - name: flambda2_debug_runtime5
             config: --enable-middle-end=flambda2 --enable-runtime5
             os: ubuntu-latest
+            build_ocamlparam: ''
```
Comment (riaqn) [inline .github/workflows/build.yml:23]: Do we need to specify this line? the macos target doesn't have this line.

**Review by riaqn — APPROVED** [review]
LGTM with a minor comment that can be ignored.

File: .github/workflows/build.yml (hunk 1 above)
Comment (mshinwell) [reply .github/workflows/build.yml:23]: It's maybe better for clarity, I copied it from the one below.

**Pull request merged**